### PR TITLE
specify Matrix::crossproduct()

### PR DESCRIPTION
**Pull request by lgatto**

### CODE REVIEW

**Comment by lgatto** [comment]
`Matrix` being a recommended package, the `requireNamespace()` is probably not needed, or is it?

**Comment by jorainer** [comment]
No, since it's in `Suggested` I would not use `requireNamespace` but call it like you did with `Matrix::` (I would add `requireNamespace` only if it was in `Enhances`).

File: R/aggregate_by_matrix.R [hunk 1]
```diff
@@ -8,7 +8,7 @@ colMeansMat <- function(x, MAT)
 #'
 #' @export
 colSumsMat <- function(x, MAT)
-  crossprod(MAT, x)
+  Matrix::crossprod(MAT, x)
```
Comment (jorainer) [inline R/aggregate_by_matrix.R:11]: have you compared performance between `Matrix::crossprod` and `crossprod` on a simple `matrix`? If there is a difference I would maybe call it only if `MAT` is a `Matrix`.

Comment (lgatto) [reply R/aggregate_by_matrix.R:11]: Using a sparse matrix is much faster, and comparable results for `Matrix::crossproduct()` and `base::crossproduct()` on a `matrix`.

```
> library(Matrix)
> library(microbenchmark)
> adjM <- sparseMatrix(i = c(sample(1:900, 100, replace = TRUE), 1000),
+                     j = c(100, sample(99, 100, replace = TRUE)),
+                     dimnames = list(paste0("Pep", 1:1000),
+                                     paste0("Prot", 1:100)))
> adjm <- as.matrix(adjM)
> m <- matrix(rnorm(1000 * 10, 10, 4),
+             nrow = 1000,
+             dimnames = list(paste0("Pep", 1:1000),
+                             paste0("S", 1:10)))
> microbenchmark(Matrix::crossprod(adjM, m),
+                Matrix::crossprod(adjm, m),               
+                base::crossprod(adjm, m))
+ 
+ Unit: microseconds
                       expr      min        lq       mean   median       uq
 Matrix::crossprod(adjM, m)   31.912   34.9755   72.71188   38.952   44.601
 Matrix::crossprod(adjm, m) 1061.765 1282.4905 1325.09600 1292.389 1301.442
   base::crossprod(adjm, m) 1056.330 1279.0145 1311.59744 1288.577 1297.289
      max neval cld
 3072.889   100  a 
 4179.681   100   b
 4093.303   100   b
```

Comment (jorainer) [reply R/aggregate_by_matrix.R:11]: Great! Well, then I'm totally fine using `Matrix::crossprod`.

**Review by jorainer — APPROVED** [review]
I'm fine with the PR - assuming you addressed @sgibb's comments.

**Pull request merged**